src/main/java/uk/gov/hmcts/reform/sscs/services/CaseLoaderService.java

```diff
@@ -42,15 +42,14 @@ public CaseLoaderService(SftpSshService sftpSshService, XmlValidator xmlValidato
 
     public void process() {
         log.info("*** case-loader *** Reading xml files from SFTP...");
-        List<GapsInputStream> inputStreamList = sftpSshService.readExtractFiles();
+        List<Gaps2File> files = sftpSshService.getFiles();
         log.info("*** case-loader *** Read xml files from SFTP successfully");
-        for (GapsInputStream gapsInputStream : inputStreamList) {
-            try (InputStream inputStream = gapsInputStream.getInputStream()) {
+        for (Gaps2File file : files) {
+            try (InputStream inputStream = sftpSshService.readExtractFile(file)) {
                 String xmlAsString = fromInputStreamToString(inputStream);
-                String type = gapsInputStream.getIsDelta() ? "Delta" : "Reference";
-                xmlValidator.validateXml(xmlAsString, type);
-                log.info("*** case-loader *** Validate " + type + " xml file successfully");
-                if ("Delta".equals(type)) {
+                xmlValidator.validateXml(xmlAsString, file.isDelta());
+                log.info("*** case-loader *** Validate delta xml file successfully");
```

Comment:
when it iterates to validate the reference the log message will be confusing as it will say Delta, right?

Reply:
Refactor to happen next. Need to process reference file separately, so will split out exceptions then.